pulsar-client-cpp/lib/KeyValue.cc

```diff
@@ -0,0 +1,86 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include <pulsar/KeyValue.h>
+#include <pulsar/Schema.h>
+#include "SharedBuffer.h"
+
+using namespace pulsar;
+
+namespace pulsar {
+
+class PULSAR_PUBLIC KeyValueImpl {
+   public:
+    std::string keyContent_;
+    std::string valueContent_;
+    KeyValueEncodingType keyValueEncodingType;
+
+    KeyValueImpl(){};
+    KeyValueImpl(std::string key, std::string value, KeyValueEncodingType keyValueEncodingType)
+        : keyContent_(key), valueContent_(value), keyValueEncodingType(keyValueEncodingType){};
+};
+
+KeyValue::KeyValue() : impl_() {}
+
+KeyValue::KeyValue(const std::string &data, const KeyValueEncodingType &keyValueEncodingType) {
+    impl_ = std::make_shared<KeyValueImpl>();
+    impl_->keyValueEncodingType = keyValueEncodingType;
+    if (impl_->keyValueEncodingType == INLINE) {
+        SharedBuffer buffer = SharedBuffer::copy(data.c_str(), data.length());
+        int keySize = buffer.readUnsignedInt();
```

Comment:
Seems we need to handle the case of `keySize` or `valueSize` is 0. In the java client, it will return null if the keySize or valueSize is zero. But here it only returns the empty string.